core/src/main/java/org/apache/calcite/rel/rules/ProjectToWindowRule.java

```diff
@@ -165,6 +169,18 @@ public ProjectToLogicalProjectAndWindowRule(
       final CalcRelSplitter transform = new WindowedAggRelSplitter(calc,
           call.builder()) {
         @Override protected RelNode handle(RelNode rel) {
+          if (rel instanceof LogicalProject && rel.getInput(0) instanceof LogicalWindow) {
```

Comment:
If we derive the collation of LogicalWindow in https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rel/rules/ProjectToWindowRule.java#L249, this change might be not necessary.

Reply:
Good point! I will try this.